exercises/circular-buffer/test/test_circular_buffer.c

```diff
@@ -0,0 +1,310 @@
+
+   circular_buffer_t *buffer = new_circular_buffer(capacity);
+   write_values_to_buffer(2, (buffer_value_t[]) {
+                          1, 2}, false, buffer);
+   read_values_from_buffer(1, (buffer_value_t[]) {
+                           1}, buffer);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          3}, false, buffer);
+   read_values_from_buffer(2, (buffer_value_t[]) {
+                           2, 3}, buffer);
+
+   delete_buffer(buffer);
+}
+
+static void test_cleared_buffer_cannot_be_read(void)
+{
+   TEST_IGNORE();
+   size_t capacity = 1;
+   buffer_value_t read_value = 0;
+
+   circular_buffer_t *buffer = new_circular_buffer(capacity);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          1}, false, buffer);
+   clear_buffer(buffer);
+   int16_t status = read(buffer, &read_value);
+   TEST_ASSERT_EQUAL_INT16(EXIT_FAILURE, status);
+   TEST_ASSERT_EQUAL_INT16(ENODATA, errno);
+
+   delete_buffer(buffer);
+}
+
+static void test_clear_frees_capacity_for_another_write(void)
+{
+   TEST_IGNORE();
+   size_t capacity = 1;
+
+   circular_buffer_t *buffer = new_circular_buffer(capacity);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          1}, false, buffer);
+   clear_buffer(buffer);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          2}, false, buffer);
+   read_values_from_buffer(1, (buffer_value_t[]) {
+                           2}, buffer);
+
+   delete_buffer(buffer);
+}
+
+static void test_clear_does_nothing_on_an_empty_buffer(void)
+{
+   TEST_IGNORE();
+   size_t capacity = 1;
+   buffer_value_t values[] = { 1 };
+   size_t values_length = ARRAY_LENGTH(values);
+
+   circular_buffer_t *buffer = new_circular_buffer(capacity);
+   clear_buffer(buffer);
+   write_values_to_buffer(values_length, values, false, buffer);
+
+   delete_buffer(buffer);
+}
+
+static void test_overwrite_acts_like_write_on_non_full_buffer(void)
+{
+   TEST_IGNORE();
+   size_t capacity = 2;
+
+   circular_buffer_t *buffer = new_circular_buffer(capacity);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          1}, false, buffer);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          2}, true, buffer);
+   read_values_from_buffer(2, (buffer_value_t[]) {
+                           1, 2}, buffer);
+
+   delete_buffer(buffer);
+}
+
+static void test_overwrite_replaces_oldest_item_on_full_buffer(void)
+{
+   TEST_IGNORE();
+   size_t capacity = 2;
+
+   circular_buffer_t *buffer = new_circular_buffer(capacity);
+   write_values_to_buffer(2, (buffer_value_t[]) {
+                          1, 2}, false, buffer);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          3}, true, buffer);
+   read_values_from_buffer(2, (buffer_value_t[]) {
+                           2, 3}, buffer);
+
+   delete_buffer(buffer);
+}
+
+static void test_overwrite_replaces_oldest_item_reamining_following_read(void)
```

Comment:
```suggestion
static void test_overwrite_replaces_oldest_item_remaining_following_read(void)
```

Reply:
D'Oh! IDE is supposed to warn me about typos like this